src/podofo/private/FreetypePrivate.cpp

```diff
@@ -195,6 +195,9 @@ bool tryExtractDataFromTTC(FT_Face face, charbuff& buffer)
     size = numFonts * sizeof(uint32_t);
     rc = FT_Load_Sfnt_Table(face, 0, sizeof(TTCF_Header), (FT_Byte*)offsets.data(), &size);
     CHECK_FT_RC(rc, FT_Load_Sfnt_Table);
+    if (face->face_index < 0 || static_cast<size_t>(face->face_index) >= offsets.size())
+        return false;
```

Comment:
I had an invalid input file that runs into this situation. Assertions are not sufficient, because they do not protect release builds against attacks using such malicious input files.

I have not kept this particular input file, but can we agree that the TTC font, coming from the pdf file, should not be considered trusted data?

Reply:
I would really have loved to have a look at the failing case. I'm unsure if PDF can carry TTC fonts, that function was made for user loading specific faces in a TTC fonts. If you find the TTC font (or PDF) that had this problem please attach it here as they may be bugs somewhere else.